drivers/gpio/gpio_nrfx.c

```diff
@@ -0,0 +1,467 @@
+	nrf_gpio_cfg_sense_set(abs_pin, NRF_GPIO_PIN_NOSENSE);
+
+	int res = 0;
+
+	/* Pins trigger interrupts only if pin has been configured to do so
+	 * and callback has been enabled for that pin.
+	 */
+	if ((data->pin_int_en & BIT(pin)) && (data->int_en & BIT(pin))) {
+		if (data->trig_edge & BIT(pin)) {
+		/* For edge triggering we use GPIOTE channels. */
+			nrf_gpiote_polarity_t pol;
+
+			if (data->double_edge & BIT(pin)) {
+				pol = NRF_GPIOTE_POLARITY_TOGGLE;
+			} else if (((data->active_level & BIT(pin)) != 0)
+				   ^ ((BIT(pin) & data->inverted) != 0)) {
+				pol = NRF_GPIOTE_POLARITY_LOTOHI;
+			} else {
+				pol = NRF_GPIOTE_POLARITY_HITOLO;
+			}
+
+			res = gpiote_channel_alloc(abs_pin, pol);
+		} else {
+		/* For level triggering we use sense mechanism. */
+			nrf_gpio_pin_sense_t sense;
+
+			if (((data->active_level & BIT(pin)) != 0)
+			    ^ ((BIT(pin) & data->inverted) != 0)) {
+				sense = NRF_GPIO_PIN_SENSE_HIGH;
+			} else {
+				sense = NRF_GPIO_PIN_SENSE_LOW;
+			}
+
+			nrf_gpio_cfg_sense_set(abs_pin, sense);
+		}
+	}
+	return res;
+}
+
+static int gpio_nrfx_config(struct device *port, int access_op,
+			    u32_t pin, int flags)
+{
+	struct gpio_nrfx_data *data = get_port_data(port);
+	nrf_gpio_pin_pull_t pull;
+	nrf_gpio_pin_drive_t drive;
+	nrf_gpio_pin_dir_t dir;
+	nrf_gpio_pin_input_t input;
+	u8_t from_pin;
+	u8_t to_pin;
+
+	switch (flags & (GPIO_DS_LOW_MASK | GPIO_DS_HIGH_MASK)) {
+	case GPIO_DS_DFLT_LOW | GPIO_DS_DFLT_HIGH:
+		drive = NRF_GPIO_PIN_S0S1;
+		break;
+	case GPIO_DS_DFLT_LOW | GPIO_DS_ALT_HIGH:
+		drive = NRF_GPIO_PIN_S0H1;
+		break;
+	case GPIO_DS_DFLT_LOW | GPIO_DS_DISCONNECT_HIGH:
+		drive = NRF_GPIO_PIN_S0D1;
+		break;
+
+	case GPIO_DS_ALT_LOW | GPIO_DS_DFLT_HIGH:
+		drive = NRF_GPIO_PIN_H0S1;
+		break;
+	case GPIO_DS_ALT_LOW | GPIO_DS_ALT_HIGH:
+		drive = NRF_GPIO_PIN_H0H1;
+		break;
+	case GPIO_DS_ALT_LOW | GPIO_DS_DISCONNECT_HIGH:
+		drive = NRF_GPIO_PIN_H0D1;
+		break;
+
+	case GPIO_DS_DISCONNECT_LOW | GPIO_DS_DFLT_HIGH:
+		drive = NRF_GPIO_PIN_D0S1;
+		break;
+	case GPIO_DS_DISCONNECT_LOW | GPIO_DS_ALT_HIGH:
+		drive = NRF_GPIO_PIN_D0H1;
+		break;
+
+	default:
+		return -EINVAL;
+	}
+
+	if ((flags & GPIO_PUD_MASK) == GPIO_PUD_PULL_UP) {
+		pull = NRF_GPIO_PIN_PULLUP;
+	} else if ((flags & GPIO_PUD_MASK) == GPIO_PUD_PULL_DOWN) {
+		pull = NRF_GPIO_PIN_PULLDOWN;
+	} else {
+		pull = NRF_GPIO_PIN_NOPULL;
+	}
+
+	dir = ((flags & GPIO_DIR_MASK) == GPIO_DIR_OUT)
+	      ? NRF_GPIO_PIN_DIR_OUTPUT
+	      : NRF_GPIO_PIN_DIR_INPUT;
+
+	input = (dir == NRF_GPIO_PIN_DIR_INPUT)
+		? NRF_GPIO_PIN_INPUT_CONNECT
+		: NRF_GPIO_PIN_INPUT_DISCONNECT;
+
+	if (access_op == GPIO_ACCESS_BY_PORT) {
+		from_pin = 0;
+		to_pin   = 31;
+	} else {
+		from_pin = pin;
+		to_pin   = pin;
+	}
+
+	for (u32_t curr_pin = from_pin; curr_pin <= to_pin; ++curr_pin) {
+		nrf_gpio_cfg(
+			NRF_GPIO_PIN_MAP(get_port_cfg(port)->port_num,
+					 curr_pin),
+			dir, input, pull, drive, NRF_GPIO_PIN_NOSENSE);
```

Comment:
Such indentation is inconsistent with what you've got right below. IMO it would be better to write it this way:
```
		nrf_gpio_cfg(NRF_GPIO_PIN_MAP(get_port_cfg(port)->port_num,
					      curr_pin),
			     dir, input, pull, drive, NRF_GPIO_PIN_NOSENSE);
```